pkg/cover/canonicalizer.go

```diff
@@ -59,7 +60,12 @@ func (can *Canonicalizer) NewInstance(modules []host.KernelModule) *Canonicalize
 	instToCanonicalMap := make(map[uint32]*canonicalizerModule)
 	canonicalToInstMap := make(map[uint32]*canonicalizerModule)
 	for _, module := range modules {
-		canonicalAddr := can.modules[module.Name]
+		canonicalAddr, found := can.modules[module.Name]
+		if !found {
+			log.Errorf("instance's modules differ from canonical. Disabling canonicalization")
+			return &CanonicalizerInstance{canonical: *can}
```

Comment:
Won't there be a problem if some instances do canonicalization and some not?

Reply:
It would be a problem; without canonicalization that instance's coverage for modules will not be accurate. However, this case only appears when the modules have changed, which will only happen when a build is changed during fuzzing. The only case I can think of is if a non-persistent build is flashed, causing a revert to an older build upon reboot.

The other option I can think of is completely stopping fuzzing with this error, instead of just disabling canonicalization.

Reply:
The main reason why you needed to add PC canonicalization was that syzkaller expects that all VMs have the same PC addresses and therefore it can maintain some global deduplicated corpus. If VMs return different PCs for same programs (and, even worse, if they return different PCs each time VM is restarted), the corpus will just explode, fuzzing efficiency drop, coverage report stop making any sense.

So if it happened that some VMs are running one image and other VMs another kernel image and we have noticed it (e.g. by discovering that module sets are different), I think we should just abort fuzzing with `log.Fatalf` so that the person maintaining the setup would immediately notice.

Reply:
That makes sense, thanks. I've updated it to just abort immediately